umm-spec-lib/src/cmr/umm_spec/umm_to_xml_mappings/dif9.clj

```diff
@@ -38,6 +38,29 @@
   [platforms]
   (generate-short-name-long-name-elements :Source_Name platforms))
 
+(defn generate-dataset-citation
+  "Returns the dif9 Data_Set_Citations from UMM-C."
+  [c]
+  (if (empty? (:CollectionCitations c))
+    [:Data_Set_Citation
+     [:Version (:Version c)]
+     [:Dataset_DOI (get-in c [:DOI :DOI])]]
+    (for [collection-citation (:CollectionCitations c)]
+      [:Data_Set_Citation
+       [:Dataset_Creator (:Creator collection-citation)]
+       [:Dataset_Editor (:Editor collection-citation)]
+       [:Dataset_Title (:Title collection-citation)]
+       [:Dataset_Series_Name (:SeriesName collection-citation)]
+       [:Dataset_Release_Date (:ReleaseDate collection-citation)]
+       [:Dataset_Release_Place (:ReleasePlace collection-citation)]
+       [:Dataset_Publisher (:Publisher collection-citation)]
+       [:Version (:Version c)]
```

Comment:
I am of the understanding that this code is ok and that if that changes a ticket will be written later to address multiple versions.